api/jobs_test.go

```diff
@@ -57,27 +60,40 @@ func TestCreateJobHandler(t *testing.T) {
 		},
 	}
 
+	indexerMock := &apiMock.IndexerMock{
+		CreateIndexFunc: func(ctx context.Context, serviceAuthToken, searchAPISearchURL string, httpClient dpHTTP.Clienter) (*http.Response, error) {
+			resp := &http.Response{
+				StatusCode: 201,
+			}
+			return resp, nil
+		},
+		GetIndexNameFromResponseFunc: func(ctx context.Context, body io.ReadCloser) (string, error) {
+			return "ons1638363874110115", nil
+		},
+	}
+
 	Convey("Given a Search Reindex Job API that can create valid search reindex jobs and store their details in a Data Store", t, func() {
 		api.NewID = func() string { return validJobID1 }
 		cfg, err := config.Get()
 		So(err, ShouldBeNil)
-		apiInstance := api.Setup(ctx, mux.NewRouter(), dataStorerMock, &apiMock.AuthHandlerMock{}, taskNames, cfg)
-		createJobHandler := apiInstance.CreateJobHandler(ctx)
+		httpClient := dpHTTP.NewClient()
+
+		apiInstance := api.Setup(mux.NewRouter(), dataStorerMock, &apiMock.AuthHandlerMock{}, taskNames, cfg, httpClient, indexerMock)
 
 		Convey("When a new reindex job is created and stored", func() {
 			req := httptest.NewRequest("POST", "http://localhost:25700/jobs", nil)
 			resp := httptest.NewRecorder()
 
-			createJobHandler.ServeHTTP(resp, req)
+			apiInstance.CreateJobHandler(resp, req)
```

Comment:
elsewhere we appear to be calling through `ServeHTTP` - how come this one is different?

Reply:
for next time - use apiInstance.Router.ServeHTTP instead to cover routing etc.